extensions/scheduler/runtime/src/main/java/io/quarkus/scheduler/runtime/SimpleScheduler.java

```diff
@@ -288,29 +294,43 @@ public Trigger getScheduledJob(String identity) {
             this.invoker = invoker;
         }
 
-        void execute(ZonedDateTime now, ExecutorService executor) {
+        void execute(ZonedDateTime now, ExecutorService executor, Vertx vertx) {
             if (!trigger.isRunning()) {
                 return;
             }
             ZonedDateTime scheduledFireTime = trigger.evaluate(now);
             if (scheduledFireTime != null) {
-                try {
-                    executor.execute(new Runnable() {
-                        @Override
-                        public void run() {
-                            try {
-                                invoker.invoke(new SimpleScheduledExecution(now, scheduledFireTime, trigger));
-                            } catch (Throwable t) {
-                                // already logged by the StatusEmitterInvoker
+                if (invoker.isBlocking()) {
+                    try {
+                        executor.execute(new Runnable() {
+                            @Override
+                            public void run() {
+                                doInvoke(now, scheduledFireTime);
                             }
+                        });
+                    } catch (RejectedExecutionException e) {
+                        LOG.warnf("Rejected execution of a scheduled task for trigger %s", trigger);
+                    }
+                } else {
+                    Context context = VertxContext.getOrCreateDuplicatedContext(vertx);
```

Comment:
Hm, I think that the `context` will be always `null` because in case of `SimpleScheduler` the `ScheduledTask.execute()` method is always executed on a dedicated `JBossScheduledThreadPoolExecutor` thread and in case of `QuartzScheduler` the method is always invoked on a dedicated Quartz thread.

Reply:
@cescoffier In other words, it should never be invoked on a `VertxThread`.

Reply:
Perfect, the. Your current approach is correct!